main.go

```diff
@@ -141,15 +144,39 @@ func ansibleEnable() {
 }
 
 func getAnsibleRepository(runDir string) error {
-	remoteURL := fmt.Sprintf("%s://%s", viper.GetString("http-proto"), viper.GetString("http-url"))
+	httpURL := viper.GetString("http-url")
+	s3Obj := viper.GetString("s3-obj")
+	nonEmptyResourceParameters := 0
+	for _, resource := range []string{httpURL, s3Obj} {
+		if resource != "" {
+			nonEmptyResourceParameters++
+		}
+	}
+	if nonEmptyResourceParameters != 1 {
+		return errors.New("exactly one remote resource must be specified. Choose one 'http-url' or 's3-obj'")
+	}
+
 	localCacheFile := fmt.Sprintf("/tmp/%s.tgz", appName)
 
-	err := idempotentFileDownload(remoteURL, localCacheFile, viper.GetString("http-user"), viper.GetString("http-pass"))
-	if err != nil {
-		return errors.Wrap(err, "unable to pull Ansible repo")
+	if httpURL != "" {
+		remoteHTTPURL := fmt.Sprintf("%s://%s", viper.GetString("http-proto"), httpURL)
+		err := idempotentFileDownload(remoteHTTPURL, localCacheFile, viper.GetString("http-user"), viper.GetString("http-pass"))
+		if err != nil {
+			return errors.Wrap(err, "unable to pull Ansible repo")
+		}
+	}
+
+	if s3Obj != "" {
+		// If MD5 checksumming on S3 objects (to avoid re-downloading) is required in the future then
```

Comment:
maybe yes, maybe no - depends on the libraries at that point. Better to point out that it doesn't do MD5 checking at all. Also worth noting the difference in the documentation as that can have some serious surprises in terms of requests/costs.

Reply:
Removed comment... Better just let the future decide!